### ISSUE TITLE
use detach_ to cut gradient propagation ， but get a lower accuracy

### ISSUE THREAD

**Issue by qk-huang** [pytorch/pytorch#7462]
this is my code：
```
probs = (P*class_mask) 
pt = torch.pow(1-probs, 0)
pt.detach_()
logmean  = (probs.log()).sum(1).view(-1,1)
logmean  = (pt*probs.log()).sum(1).view(-1,1)
```
the two loss function above with pt or not have different test accuracy, the first function always have a higher acc than the second. I have tried many times .But in my opinion, due to the pt is identically equal to 1 and has been detached，pt will not have an impact on gradients. is there anything that I understand wrongly?thanks

cc @ezyang @albanD @zou3519 @gqchen @pearu @nikitaved

**Comment by albanD** [comment]
Hi,

We use github issues only for bugs or feature requests.
Please use the forum to ask questions: https://discuss.pytorch.org/

For your question, I cannot reproduce it with a simple script:
```python
import torch

probs = torch.rand(2, 3, requires_grad=True)
pt = torch.pow(1-probs, 0)
pt.detach_()
logmean1  = (probs.log()).sum(1).view(-1,1)
logmean2  = (pt*probs.log()).sum(1).view(-1,1)

grad1, = torch.autograd.grad(logmean1.sum(), probs)
grad2, = torch.autograd.grad(logmean2.sum(), probs)

print("Difference", (grad1 - grad2).abs().sum())
```

If you have more question, please open a post on the forum.
